tests/integration/rpath_extension/setup.py

```diff
@@ -0,0 +1,32 @@
+import subprocess
+
+from setuptools import Extension
+from setuptools import setup
+from setuptools.command.build_ext import build_ext
+
+# Compile the shared library before building the extension
+subprocess.run(
+    ["gcc", "-shared", "-o", "sharedlibs/sharedlib.so", "sharedlibs/sharedlib.c"]
+)
+
+
+class CustomBuildExt(build_ext):
+    def run(self):
+        # Run the original build_ext command
+        build_ext.run(self)
+
+        # Add code to compile the shared library
+
+
+setup(
+    name="ext",
+    version="1.0",
+    ext_modules=[
+        Extension(
+            "ext",
+            sources=["ext.c"],
+            extra_link_args=["-Wl,-rpath,$ORIGIN/sharedlibs"],
+        )
+    ],
+    cmdclass={"build_ext": CustomBuildExt},
```

Comment:
Looks like this (and `CustomBuildExt` itself) are no longer needed, since you're using a `subprocess.run()` at module scope to do the building of the .so